terraform/groups/ecs-service/variables.tf

```diff
@@ -0,0 +1,121 @@
+}
+variable "aws_profile" {
+  default     = "development-eu-west-2"
+  type        = string
+  description = "The AWS profile to use for deployment."
+}
+
+# ------------------------------------------------------------------------------
+# Docker Container
+# ------------------------------------------------------------------------------
+variable "docker_registry" {
+  type        = string
+  description = "The FQDN of the Docker registry."
+}
+
+# ------------------------------------------------------------------------------
+# Service performance and scaling configs
+# ------------------------------------------------------------------------------
+variable "desired_task_count" {
+  type = number
+  description = "The desired ECS task count for this service"
+  default = 1 # defaulted low for dev environments, override for production
+}
+variable "required_cpus" {
+  type = number
+  description = "The required cpu resource for this service. 1024 here is 1 vCPU"
+  default = 256 # defaulted low for dev environments, override for production
+}
+variable "required_memory" {
+  type = number
+  description = "The required memory for this service"
+  default = 512 # defaulted low for node service in dev environments, override for production
+}
+
+variable "max_task_count" {
+  type        = number
+  description = "The maximum number of tasks for this service."
+  default     = 3
+}
+
+variable "use_fargate" {
+  type        = bool
+  description = "If true, sets the required capabilities for all containers in the task definition to use FARGATE, false uses EC2"
+  default     = true
+}
+variable "use_capacity_provider" {
+  type        = bool
+  description = "Whether to use a capacity provider instead of setting a launch type for the service"
+  default     = true
+}
+variable "service_autoscale_enabled" {
+  type        = bool
+  description = "Whether to enable service autoscaling, including scheduled autoscaling"
+  default     = true
+}
+variable "service_autoscale_target_value_cpu" {
+  type        = number
+  description = "Target CPU percentage for the ECS Service to autoscale on"
+  default     = 50 # 100 disables autoscaling using CPU as a metric
+}
+variable "service_scaledown_schedule" {
+  type        = string
+  description = "The schedule to use when scaling down the number of tasks to zero."
+  # Typically used to stop all tasks in a service to save resource costs overnight.
+  # E.g. a value of '55 19 * * ? *' would be Mon-Sun 7:55pm.  An empty string indicates that no schedule should be created.
+
+  default     = ""
+}
+variable "service_scaleup_schedule" {
+  type        = string
+  description = "The schedule to use when scaling up the number of tasks to their normal desired level."
+  # Typically used to start all tasks in a service after it has been shutdown overnight.
+  # E.g. a value of '5 6 * * ? *' would be Mon-Sun 6:05am.  An empty string indicates that no schedule should be created.
+
+  default     = ""
+}
+
```

Comment:
need to add:
variable "service_autoscale_scale_in_cooldown" {
type = number
description = "Cooldown in seconds for ECS Service scale in (run fewer tasks)"
default = 600
}
variable "service_autoscale_scale_out_cooldown" {
type = number
description = "Cooldown in seconds for ECS Service scale out (add more tasks)"
default = 600
}